library/core/src/slice/mod.rs

```diff
@@ -3788,6 +3820,29 @@ impl<T> [T] {
     /// s.insert(idx, num);
     /// assert_eq!(s, [0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 42, 55]);
     /// ```
+    ///
+    /// If you were going to use [`binary_search`], but need specifically the
+    /// *first* matching item (as opposed to *some* matching item):
+    ///
+    /// ```
+    /// # use std::cmp::Ordering;
+    /// fn binary_search_first<T: Ord>(haystack: &[T], needle: &T) -> Result<usize, usize> {
+    ///     let low = haystack.partition_point(|x| x < needle);
+    ///     if let Some(x) = haystack.get(low) {
+    ///         if Ord::cmp(x, needle) == Ordering::Equal {
+    ///             return Ok(low);
+    ///         }
+    ///     }
+    ///
+    ///     Err(low)
+    /// }
```

Comment:
I'm not sure if we should include generic functions like this in the documentation. It feels a bit like we'd expect users to copy-paste this exact function into their code, which is probably not what we should be aiming for. Also, things like `Ord::cmp() == Equal` are probably a distraction from what this example is trying to show.

Is there a simpler example that gets the point across?

Reply:
That's fair.  I think it's covered indirectly in all the other things, so I'll just drop it from this PR.  If I come up with anything I'll send another.